multiclient/client.go

```diff
@@ -0,0 +1,531 @@
+}
+
+// Contract Calling
+
+// CallContract executes a message call transaction, which is directly executed in the VM
+// of the node, but never mined into the blockchain.
+//
+// blockNumber selects the block height at which the call runs. It can be nil, in which
+// case the code is taken from the latest known block. Note that state from very old
+// blocks might not be available.
+func (mc *Client) CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	fns := make([]getFn, len(mc.rpcClients))
+	for i, c := range mc.rpcClients {
+		ec := ethclient.NewClient(c)
+		fns[i] = func(ctx context.Context) (interface{}, error) {
+			return ec.CallContract(ctx, msg, blockNumber)
+		}
+	}
+
+	resp, err := getFromAny(ctx, fns)
+	if err != nil {
+		log.Warn("Failed to call contract from any eth client", "from", msg.From.Hex(), "to", msg.To.Hex(), "blockNumber", blockNumber.String(), "err", err)
+		return nil, err
+	}
+	hex, ok := resp.([]byte)
+	if !ok {
+		log.Error("Failed to cast type to []byte")
+		return nil, ErrInvalidTypeCast
+	}
+	return hex, nil
+}
+
+// PendingCallContract executes a message call transaction using the EVM.
+// The state seen by the contract call is the pending state.
+func (mc *Client) PendingCallContract(ctx context.Context, msg ethereum.CallMsg) ([]byte, error) {
+	fns := make([]getFn, len(mc.rpcClients))
+	for i, c := range mc.rpcClients {
+		ec := ethclient.NewClient(c)
+		fns[i] = func(ctx context.Context) (interface{}, error) {
+			return ec.PendingCallContract(ctx, msg)
+		}
+	}
+
+	resp, err := getFromAny(ctx, fns)
+	if err != nil {
+		log.Warn("Failed to call contract with pending state from any eth client", "from", msg.From.Hex(), "to", msg.To.Hex(), "err", err)
+		return nil, err
+	}
+	hex, ok := resp.([]byte)
+	if !ok {
+		log.Error("Failed to cast type to []byte")
+		return nil, ErrInvalidTypeCast
+	}
+	return hex, nil
+}
+
+// SendTransaction injects a signed transaction into the pending pool for execution.
+//
+// If the transaction was a contract creation use the TransactionReceipt method to get the
+// contract address after the transaction has been mined.
+func (mc *Client) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	fns := make([]postFn, len(mc.rpcClients))
+	for i, c := range mc.rpcClients {
+		ec := ethclient.NewClient(c)
+		fns[i] = func(ctx context.Context) error {
+			return ec.SendTransaction(ctx, tx)
+		}
+	}
+
+	err := postToAll(ctx, fns)
+	if err != nil {
+		log.Warn("Failed to send transaction to any eth client", "txHash", tx.Hash().Hex(), "err", err)
+		return err
+	}
+	return nil
+}
+
+func (mc *Client) CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error {
+	fns := make([]postFn, len(mc.rpcClients))
+	for i, c := range mc.rpcClients {
+		fns[i] = func(ctx context.Context) error {
+			return c.CallContext(ctx, result, method, args...)
+		}
+	}
+
+	err := postToAll(ctx, fns)
```

Comment:
It should be `getFromAny` if one of the rpc is successful?

Reply:
Add `isPostToAll` param to decide to perform `getFromAny` or `postToAll`.
Updated in `59b8098`.